src/grpc/getresource/mod.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::*;
+use log::*;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tonic::{transport::Server, Request, Response, Status};
+
+use crate::grpc::AGENT_NAME;
+use crate::kbc_runtime;
+use get_resource::get_resource_service_server::{GetResourceService, GetResourceServiceServer};
+use get_resource::{GetResourceRequest, GetResourceResponse};
+
+pub mod get_resource {
+    tonic::include_proto!("getresource");
+}
+
+#[derive(Debug, Default)]
+pub struct GetResource {}
+
+#[tonic::async_trait]
+impl GetResourceService for GetResource {
```

Comment:
This looks odd to me. Wouldn't `Resource` be a better name for the `struct` which would then become:

```rust
impl GetResourceService for Resource
```